src/main/kotlin/no/skatteetaten/aurora/boober/feature/HerkimerVaultFeature.kt

```diff
@@ -0,0 +1,192 @@
+                    "$FEATURE_FIELD/$key/multiple",
+                    defaultValue = false,
+                    validator = { it.boolean() })
+            )
+        }.toSet()
+    }
+
+    override fun createContext(
+        spec: AuroraDeploymentSpec,
+        cmd: AuroraContextCommand,
+        validationContext: Boolean
+    ): Map<String, Any> {
+        val herkimerVaultResources: List<HerkimerVaultResource> = spec.findAllConfiguredHerkimerResources()
+        val resources = mapOf(HERKIMER_RESOURCE_KEY to herkimerVaultResources)
+        if (validationContext) {
+            return resources
+        }
+
+        val herkimerResources: Map<HerkimerVaultResource, List<ResourceHerkimer>> =
+            herkimerVaultResources.associateWith {
+                herkimerService.getClaimedResources(spec.applicationDeploymentId, it.serviceClass)
+            }
+
+        val secrets: Map<HerkimerVaultResource, List<Secret>> = herkimerResources.map { (config, resources) ->
+            val secrets = if (!config.multiple) {
+                listOf(generateHerkimerSecret(resources.first(), spec))
+            } else {
+                resources.map {
+                    generateHerkimerSecret(it, spec)
+                }
+            }
+            config to secrets
+        }.toMap()
+
+        return resources
+            .addIfNotNull(HERKIMER_REPONSE_KEY to herkimerResources)
+            .addIfNotNull(HERKIMER_SECRETS_KEY to secrets)
+    }
+
+    override fun validate(
+        adc: AuroraDeploymentSpec,
+        fullValidation: Boolean,
+        context: Map<String, Any>
+    ): List<Exception> {
+
+        val configuredHerkimerVaultResources = context[HERKIMER_RESOURCE_KEY] as List<HerkimerVaultResource>
+
+        val errors = validateNotExistsResourcesWithMultipleTrueAndFalseWithSamePrefix(configuredHerkimerVaultResources)
+
+        if (!fullValidation) return errors
+
+        val herkimerResponses = context[HERKIMER_REPONSE_KEY] as Map<HerkimerVaultResource, List<ResourceHerkimer>>
+
+        val fullValidationErrors = herkimerResponses.filter { !it.key.multiple && it.value.size > 1 }.map {
+            IllegalStateException("Resource with key=${it.key.key} expects a single result but ${it.value.size} was returned")
+        }
+        return errors.addIfNotNull(fullValidationErrors)
+    }
+
+    override fun generate(adc: AuroraDeploymentSpec, context: Map<String, Any>): Set<AuroraResource> {
+        if (!adc.isFeatureEnabled()) return emptySet()
+
+        val herkimerVaultResources = context[HERKIMER_SECRETS_KEY] as Map<HerkimerVaultResource, List<Secret>>
+
+        return herkimerVaultResources.values.flatMap { secrets ->
+            secrets.map {
+                it.generateAuroraResource()
+            }
+        }.toSet()
+    }
+
+    private fun generateHerkimerSecret(response: ResourceHerkimer, adc: AuroraDeploymentSpec): Secret {
+
+        // TODO: Should this be first and should it be called credentials?
```

Comment:
Her kjenner jeg ikke godt til herkimer til å vite hva vi bør gjøre, men kan det være vi bør ha noe mer config for å si hvilke claims man vil ha som env vars her, og hva evt env var prefix skal være for dem?

Jeg ser ikke at ResourceClaim i herkimer har noen type heller. Må det ikke egentlig ha det?

Reply:
hmm... dersom en applicationdeployment har et claim på en resource og har credentials mot den må det jo eksponeres for applikasjonen på et vis. Så kanskje ikke nødvendig å kunne inkludere/ekskludere claims nå, siden eneste måte disse kan eksponeres til applikasjonen på er gjennom denne mekanismen?